experimental/dds/tree2/src/simple-tree/factoryInputTypes.ts

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TreeValue } from "../core";
+import { RestrictiveReadonlyRecord } from "../util";
+import {
+	FieldKind,
+	FieldKinds,
+	AllowedTypes,
+	Any,
+	FieldNodeSchema,
+	TreeFieldSchema,
+	InternalTypedSchemaTypes,
+	LeafNodeSchema,
+	MapNodeSchema,
+	ObjectNodeSchema,
+	TreeNodeSchema,
+	TreeSchema,
+} from "../feature-libraries";
+
+/**
+ * Data from which a {@link TreeObjectNode} can be built.
+ * @alpha
+ */
+export type TreeObjectNodeFactoryInput<TSchema extends ObjectNodeSchema> =
```

Comment:
Is "FactoryInput" the right name, given that it's not always input to a factory? We also use this type as the input to list insertions (we don't currently use it for map insertions, but we will need to if we ever add any custom methods to our map type). 

Even if/when we require all objects to be created by factories, `list.insert*` will still need to take this "input" type in order to support lists of lists. We need to be able to create a list directly (as an Array) that is not wrapped in an object.

We could just drop the word "Factory" here and call it `TreeObjectNodeInput`, `TreeFieldInput`, etc.

Reply:
These are used when we are constructing new nodes. The list APIs which do that implicitly are still factories  (and really should take unhydrated nodes instead so they can insert with polymorphic subtrees).

I think just "input" isn't great since lots of code using our APIs will take in nodes as inputs, and should not use these types (ex: the source for move, utilities for waking trees, app code that takes in a tree etc).

Reply:
"The list APIs which do that implicitly are still factories" - I disagree. A factory is a well defined kind of object in our library (`TreeObjectFactory`) and it is not necessarily involved when inserting content into a list. If I have a `List<List<number>>`, I insert content into it by doing `list.insert([0, 1, 2])`. No factories here, but I need the "input" type, whatever we call it.

If you don't like "*Input", what do you think of "Raw*" (e.g. `RawTreeObjectNode`)? It's short, and while the name "raw" might require explanation, I think once it's been explained it is intuitive and memorable.

Reply:
I don't like "raw". I feel like there should be a way to get raw nodes from the tree. Ex: removing the proxy wrapper gives me the "raw" flex tree node.

I think Input is better than raw, and and you raise some good points regarding how they are used without factories in some edge cases (Logically one could consider it to be run through a factory internally in those cases, and list.insert does take types which sometimes do have to be run through factories), but thats not very intuitive in the leaf cases. I'll just go with Input.

Reply:
Actually, which of the renamed types ever occur outside of a factory context? Its only TreeNodeUnionFactoryInput, and that type does already result in actual node objects getting built, and in all the non-leaf cases it goes through TreeFactories of some type (and even leaves go through a factory like thing internally).

TreeNodeObjectFctoryInput does always go through TreeObjectFactory so that naming seems fine, and non object inputs go through other kinds of factories producing nodes.

Also, since these types lack type information, they are not actually a good choice to use for unions, and as the union version is the only one used in the API (its used in List, and thats it) I think these really should be factory types, and the union one should be deprecated and replaced with an unhydrated node, which in the leaf case could just be the value.

Reply:
The list and map insertion code paths (Which are the only public consumers of these types in the tree API) produce "ExtractedFactoryContent" from these inputs. That makes the argument that these types don't have to do with factories seem confusing.

Reply:
`TreeNodeUnion` is the one used for inserting into lists (and _could_ be used for setting the values of maps) so it does not necessarily go through a tree factory. All inputs do pass through a function called `extractFactoryContent()`, but that's not saying that all inputs are "factory content", rather it means "from the content, extract anything that is from a factory" (which might be nothing). 

`TreeNodeObjectFactoryInput` is only used at the top-level, _directly_, in factories, but it's still used recursively by `TreeNodeUnionFactoryInput`. If `TreeNodeUnionFactoryInput` is called `TreeNodeUnionInput`, it's weird that some part of its subtree might be a `TreeNodeUnionFactoryInput`, because it's still not being passed to a factory.

Besides, what if we do end up reversing our (still not set-in-stone) decision that we _have_ to construct objects via their factories? Then the "Factory" type names would make even less sense. I think decoupling our "insertable content" format from the concept of factories is just simpler. Maybe that's an idea for a name - "Insertable\*" (e.g. `InsertableTreeNodeObject`).

I don't understand your last point about replacing the union type with an unhydrated node. We can't make unhydrated lists right now. We need a way to be able to insert lists into lists.

Reply:
Renamed to insertable. Added comment about how to disambiguate types